### PR TITLE
JAMES-1746 Proof of concept with junit-contracts

### PR DESCRIPTION
**Pull request by chibenwa**
Before going further, I want to have some feedbacks on basic examples. Here I want us to review the style. Do you feel like it is better than simply extending an abstract class?

Some questions I have though : 
 - How can I ignore some tests in some given implementations ? (@Ignore won't work. I am thinking to Maildir and NAMESPACE handling that fails and should be ignored. Systematically check capabilities ?)
 - I guess several test classes can use the same ContractImpl, achieving the genericity required for instance for MailboxManager tests + MPT tests.

### CODE REVIEW

**Comment by mbaechler** [comment]
That looks great.
About ignoring tests based on capabilites, there's junit Assume that match the concept very well.
I'd like to see a test case with a @Rule to see how it plays with junit-contract.

File: server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/access/CassandraAccessTokenRepositoryTest.java [hunk 1]
```diff
@@ -20,22 +20,40 @@
...
+
+    @Contract.Inject
+    public IProducer<CassandraAccessTokenRepository> getProducer() {
+        return producer;
     }
 
     @After
     public void tearDown() {
-        cassandra.clearAllTables();
+
```
Comment (rouazana) [inline server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/access/CassandraAccessTokenRepositoryTest.java]: You should be able to remove the tearDown method now

Comment (chibenwa) [inline server/data/data-jmap-cassandra/src/test/java/org/apache/james/jmap/cassandra/access/CassandraAccessTokenRepositoryTest.java]: True

**Comment by chibenwa** [comment]
I will continue the PoC with integration testing !

**Pull request closed without merging**